api/http/handler/endpointgroups/endpointgroup_endpoint_delete.go

```diff
@@ -42,5 +42,15 @@ func (handler *Handler) endpointGroupDeleteEndpoint(w http.ResponseWriter, r *ht
 		return &httperror.HandlerError{http.StatusInternalServerError, "Unable to persist endpoint changes inside the database", err}
 	}
 
+	unassignedGroup, err := handler.EndpointGroupService.EndpointGroup(portainer.EndpointGroupID(1))
```

Comment:
I would relocate this logic into the `updateEndpointRelations` function.

E.g. pass a nil parameter here

```
	err = handler.updateEndpointRelations(*endpoint, nil)
	if err != nil {
		return &httperror.HandlerError{http.StatusInternalServerError, "Unable to persist endpoint relations changes inside the database", err}
	}
```

And if the endpoint group is nil in the `updateEndpointRelations` function, lookup the unassigned group. See my comment below regarding the use of pointers.